### DISCUSSION TITLE
systemd-resolved versus avahi for dnssd

### DISCUSSION THREAD

**Discussion by gearhead** [mikebrady/shairport-sync#1786] [General]
I have been building shairport-sync for a long time. I recently switched over to systemd-resolved and ditched avahi-daemon and am trying to remove dependencies on avahi. Right now, my 'configure statement' calls for what appears as both:
```
     --with-avahi \
     --with-dns_sd \
```

Can I remove the `--with-avahi` and remove the dependencies on the avahi library and will it work with dnssd as implemented in systemd-resolved? When I do so, will the airplay2 work or not? From the docs, it appears that it will not. Why not? Could it work with a patch?

**Comment by mikebrady** [comment]
Thanks for the post, and apologies for the delay. The AirPlay 2 implementation uses specific features of Avahi, so `--with-avahi` is mandatory for it.
